modules/navier_stokes/src/fvbcs/NSFVOutflowTemperatureBC.C

```diff
@@ -0,0 +1,75 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NSFVOutflowTemperatureBC.h"
+
+registerMooseObject("NavierStokesApp", NSFVOutflowTemperatureBC);
+
+InputParameters
+NSFVOutflowTemperatureBC::validParams()
+{
+  InputParameters params = FVFluxBC::validParams();
+  params.addClassDescription("Outflow velocity temperature advection boundary conditions for "
+                             "finite volume method allowing for thermal backflow.");
+  params.addParam<MooseFunctorName>(
+      "advected_quantity", "rho_cp_temp", "The heat quantity to advect.");
+  params.addRequiredParam<MooseFunctorName>("u", "The velocity in the x direction.");
+  params.addParam<MooseFunctorName>("v", "The velocity in the y direction.");
+  params.addParam<MooseFunctorName>("w", "The velocity in the z direction.");
+  params.addRequiredParam<MooseFunctorName>("backflow_T",
+                                            "The backflow temperature entering the domain.");
+  return params;
+}
+
+NSFVOutflowTemperatureBC::NSFVOutflowTemperatureBC(const InputParameters & parameters)
+  : FVFluxBC(parameters),
+    _adv_quant(getFunctor<ADReal>("advected_quantity")),
+    _u(getFunctor<ADReal>("u")),
+    _v(isParamValid("v") ? &getFunctor<ADReal>("v") : nullptr),
+    _w(isParamValid("w") ? &getFunctor<ADReal>("w") : nullptr),
+    _backflow_T(getFunctor<ADReal>("backflow_T")),
+    _dim(_subproblem.mesh().dimension())
+{
+  if (_dim >= 2 && !_v)
+    mooseError(
+        "In two or more dimensions, the v velocity must be supplied using the 'v' parameter");
+  if (_dim >= 3 && !_w)
+    mooseError("In threedimensions, the w velocity must be supplied using the 'w' parameter");
+}
+
+ADReal
+NSFVOutflowTemperatureBC::computeQpResidual()
+{
+  const auto boundary_face = singleSidedFaceArg();
+  const auto state = determineState();
+
+  ADRealVectorValue v(_u(boundary_face, state));
+  if (_v)
+    v(1) = (*_v)(boundary_face, state);
+  if (_w)
+    v(2) = (*_w)(boundary_face, state);
+
+  auto vol_flux = v * _normal;
+
+  if (vol_flux > 0)
+  {
+    auto face_arg =
+        makeFace(*_face_info, Moose::FV::LimiterType::Upwind, MetaPhysicL::raw_value(vol_flux) > 0);
+    auto rho_cp_face = _adv_quant(face_arg, determineState());
+    return rho_cp_face * _var.getBoundaryFaceValue(*_face_info, determineState()) * vol_flux;
+  }
+  else
+  {
+    auto face_arg =
+        makeFace(*_face_info, Moose::FV::LimiterType::Upwind, MetaPhysicL::raw_value(vol_flux) < 0);
+    auto rho_cp_face = _adv_quant(face_arg, determineState());
+    auto backflow_T = _backflow_T(face_arg, determineState());
+    return rho_cp_face * backflow_T * vol_flux;
+  }
```

Comment:
So the default functor is rho * cp * T. And then you are multiplying by the variable value, which is the temperature. So by default you will have the square of temperature. Also when composing `face_arg` I think we should think carefully about how we want to compose it. I would kind of think we should
1. just use `boundary_face`
2. think harder about what we supply for the third argument to `makeFace`. The part in the `if` branch is fine ... that should upwind any variable values used in the evaluation of the `_adv_quant` functor. But in the `else` branch we are being implicitly asked to provide "upwind" values for `_adv_quant` when we are outside the bounds of the domain

Reply:
Thanks. Changed to `boundary_face`. Results didn't change for the test

Reply:
Also changed the 'rho * cp * T' functor for 'rho' and 'cp'